### PR TITLE
[WIP] Use `vtex.rewriter` for sitemap data

### PR DESCRIPTION
**Pull request by vcalasans**

### CODE REVIEW

File: node/clients/rewriter.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        }
+      )
+      .then(path(['data', 'internal', 'indexFiles']) as any) as Promise<RouteIndexFiles>
+
+  public listInternals = (from: number, to: number, indexFile: string): Promise<Internal[]> =>
+    this.graphql
+      .query<Internal[], { from: number; to: number, indexFile: string }>(
+        {
+          query: `
+      query ListRedirects($from: Int!, $to: Int!, $indexFile: String) {
```
Comment (Jeymisson) [inline node/clients/rewriter.ts]: `ListInternals` ?

File: node/middlewares/newSitemap.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      <sitemap>
+        <loc>https://${forwardedHost}${rootPath}/newsitemap/${entity}-${firstIndex}-${lastIndex}.xml</loc>
+        <lastmod>${currentDate()}</lastmod>
+      </sitemap>
+    `
+  }
+  return `
+    <sitemap>
+      <loc>https://${forwardedHost}${rootPath}/newsitemap/${entity}.xml</loc>
+      <lastmod>${currentDate()}</lastmod>
```
Comment (Jeymisson) [inline node/middlewares/newSitemap.ts]: Can't we read this from rewritter indexing date? What is `endDate` value for?

File: node/middlewares/newSitemap.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    </sitemap>
+  `
+}
+
+const URLEntry = (forwardedHost: string, rootPath: string, route: Internal): string => {
+  let entry = `
+      <loc>https://${forwardedHost}/${rootPath}${route.from}</loc>
+      <lastmod>${currentDate()}</lastmod>
```
Comment (Jeymisson) [inline node/middlewares/newSitemap.ts]: Same https://github.com/vtex-apps/store-sitemap/pull/35/files#r332740140

File: node/middlewares/newSitemap.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    indexTitles.forEach((entity: string) => {
+      const indexSize = indexTable[entity]
+      if (indexSize <= MAX_ROUTES_PER_REQUEST) {
+        $('sitemapindex').append(sitemapIndexEntry(forwardedHost, rootPath, entity))
+      } else {
+        const indexBatches = map(
+          x => [x*MAX_ROUTES_PER_REQUEST, (x+1)*MAX_ROUTES_PER_REQUEST-1],
+          range(0, Math.floor(indexSize/MAX_ROUTES_PER_REQUEST))
+        )
+        indexBatches.forEach(([firstIndex, lastIndex]) =>
```
Comment (Jeymisson) [inline node/middlewares/newSitemap.ts]: We could make only one loop:
```const indexRange = range(0, Math.floor(indexSize/MAX_ROUTES_PER_REQUEST)
indexBatches.forEach(range => {
   const [firstIndex, lastIndex] = [range*MAX_ROUTES_PER_REQUEST, (range+1)*MAX_ROUTES_PER_REQUEST-1]
})
   $('sitemapindex').append(sitemapIndexEntry(forwardedHost, rootPath, entity, firstIndex, lastIndex)))
```

File: node/clients/rewriter.ts (hunk 1 above)
Comment (vcalasans) [reply node/clients/rewriter.ts]: Tks!

File: node/middlewares/newSitemap.ts (hunk 2 above)
Comment (vcalasans) [reply node/middlewares/newSitemap.ts]: True. tks!

File: node/middlewares/newSitemap.ts (hunk 4 above)
Comment (vcalasans) [reply node/middlewares/newSitemap.ts]: True! Tkssss

**Pull request merged**